src/Features/Core/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -291,7 +291,7 @@ private bool CheckOption(Workspace workspace, string language, bool documentOpen
         {
             try
             {
-                if (!skipClosedFileChecks && !CheckOption(project.Solution.Workspace, project.Language, documentOpened: project.Documents.Any(d => d.IsOpen())))
+                if (!skipClosedFileChecks && !CheckOption(project.Solution.Workspace, project.Language, documentOpened: false))
```

Comment:
If that is the case then any action can generate diagnostic in any document, I dont understand the bias behind running certain independent actions and not other. I think semantics of flag should be changed to local open file diagnostics only OR all diagnostics that can be run via running independent actions that dont need force computing closed file diagnostics. I have been assuming the later, and I think you are meaning the former.